runtime/onert/core/src/compiler/pass/ConstantOutputPass.cc

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ConstantOutputPass.h"
+
+#include "ir/Graph.h"
+#include "ir/operation/Permute.h"
+#include "util/logging.h"
+
+namespace onert
+{
+namespace compiler
+{
+namespace pass
+{
+
+void ConstantOutputPass::callback(const ir::OperandIndex &ind, ir::Operand &obj)
+{
+  if (!_graph.getOutputs().contains(ind) || !obj.isConstant())
+    return;
+
+  auto permute_input_ind = _graph.addOperand(obj.shape(), obj.typeInfo());
+  auto &permute_input_obj = _graph.operands().at(permute_input_ind);
+
+  // Move the const data
+  permute_input_obj.data(obj.shareData());
+  obj.releaseData();
+  obj.info().setAsNonConst();
+
+  using ir::operation::Permute;
+  auto permute_obj = std::make_unique<Permute>(permute_input_ind, ind, Permute::Type::COPY);
+  auto permute_ind = _graph.operations().push(std::move(permute_obj));
+
+  permute_input_obj.insertUse(permute_ind);
+  obj.setDef(permute_ind);
```

Comment:
From https://github.com/Samsung/ONE/pull/4334/files#r492524848

It seems to make the graph like below
```
== Original model ==
(( Const Output )) 
         \-> [ Some Operation on cpu ] -> ( Some Tensor ) -> ...
         \-> [ Some Operation on acl_cl ] -> ( Some Tensor ) -> ...

== After ConstantOutputPass ==
( Const ) -> [ Permute ] -> (( Output (Non-const) ))  -> [ Some Operation on cpu ] -> ( Some Tensor ) -> ...
                                                      \-> [ Some Operation on acl_cl ] -> ( Some Tensor ) -> ...

== After ConstantInsertionPass ==
Nothing. "Output" is not const.

== After PermutationInsertionPass ==
( Const ) -> [ Permute ] -> (( Output (Non-const) ))  -> [ Some Operation on cpu ] -> ( Some Tensor ) -> ...
                                                      \->  [ Permute ] -> ( Permuted Tensor )  -> [ Some Operation on acl_cl ] -> ( Some Tensor ) -> ...
```

Is it right? If so, please do not change operation's const operand to non-const. There are some problems as follows:
1. Those operations using "Output" cannot get constant data from the "Output" operand in KernelGenerator. In model, the operand is constant and has data, but in graph after running passes, the operand doen't have data.
2. Some kernels may cache constant data with transposing it for performance. But if they cannot cache non-constant data.
3. Performance will be decrease by that increased "Permute" node .

Reply:
This problems would be fixed by @wateret in other PR.

Reply:
@ragmani Thank you for detailed review 😄